### ISSUE TITLE
[Bug] Sometimes gradient doesn't back-propagate after view

### ISSUE THREAD

**Issue by ShangtongZhang** [pytorch/pytorch#8474]
## Issue description

Gradient doesn't back-propagate after `view`

## Code example

```
import torch

x1 = torch.rand(5).requires_grad_()
x1 = x1.view((-1, 1))
x1.sum().backward()
print(x1.grad)

x1 = torch.rand(5).requires_grad_()
x1.sum().backward()
print(x1.grad)
```

Output

```
None
tensor([ 1.,  1.,  1.,  1.,  1.])
```
## System Info

```
(Python3.6) Shangtong@Shangtong:~/Downloads$ python collect_env.py
Collecting environment information...
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.12.6
GCC version: Could not collect
CMake version: version 3.10.1

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip3] numpy (1.14.3)
[pip3] torch (0.4.0)
[pip3] torchvision (0.2.1)
[conda] Could not collect
```

**Comment by fmassa** [comment]
I don't think this is a bug. He is checking the gradient of a non-leaf variable.
Change the name of the viewed variable to `x2`, and verify that the gradient of `x1` is good.

**Comment by SsnL** [comment]
Yes you are correct. @fmassa My bad.
